pkg/server/server.go

```diff
@@ -383,31 +386,32 @@ func (s *Server) VerifyEmail(c *gin.Context, redirectURL string) {
 	if !exists {
 		s.redirectError(c, errors.ErrInvalidAccessToken)
 		// Redirecting user to UI if the user is not authorized
-		c.Redirect(http.StatusUnauthorized, redirectURL)
+		c.Redirect(http.StatusPermanentRedirect, redirectURL)
 		return
 	}
 	jwtUserCredentials := jwtUser.(*models.UserCredentials)
 
 	if jwtUserCredentials.UnverifiedEmail != nil {
-		*jwtUserCredentials.Email = jwtUserCredentials.GetUnverifiedEmail()
-		*jwtUserCredentials.UnverifiedEmail = ""
+		tmp := jwtUserCredentials.GetUnverifiedEmail()
+		jwtUserCredentials.Email = &tmp
+		jwtUserCredentials.UnverifiedEmail = nil
```

Comment:
What are we doing in the above 3 lines?

Reply:
Actually in the removed line number 392, we were  trying to set `Email` but its giving nil pointer exception as the `Email` can be empty.
So here we extract the `unverfied_email` (as we can't set it directly since its a pointer, dereferencing it afterwards will cause problems), and set it in the `email` field then making the `unverified_email` empty

Reply:
Why do we need a string pointer instead of just a string? We don't need to worry about having nil, we can have as empty string.

Reply:
OK we can change these types to string only. We can take this part in a separate PR.